Biallelic filtering and bcftools comparison

briankirz opened the pull request:
biallelic.py filters out all non-biallelic snps from a gzipped vcf file and writes them to a new vcf file.

compare_vcf confirms its accuracy with another output file resulting from a bcftools call to do the same.

briankirz committed:
biallelic filtering

biallelic.py filters out all non-biallelic snps from a gzipped vcf file and writes them to a new vcf file.

compare_vcf confirms its accuracy with another output file resulting from a bcftools call to do the same.

David-Peede reviewed (changes requested):
- [x]  Add the `bcftools` command in a shell script and annotate
- [x]  Annotate every line of your python functions
- [x] Incorporate the `INFO` field's `VT` flag
- [x] Add a README.MD File briefly explaining (one sentence) what each script in this directory does

briankirz committed:
Added comments and changes

Added bcf shell script
Added readme
Included VT=SNP in biallelic.py

David-Peede reviewed (changes requested):
- [x] make the README.MD formatted as a markdown file
- [x] write the bcf_biallelic.sh as shell file that you will submit to oscar (if you get stuck look at my example [here](https://github.com/David-Peede/BIOL1435/blob/main/OSCAR/get1KsampleFile.sh) or at the [documentation](https://docs.ccv.brown.edu/oscar/submitting-jobs/shared-machine))
- [x] look at my `tgp_site_info` function starting on line 200 from [here](https://github.com/David-Peede/targt_pipeline_vcf/blob/master/vcf_functions.py) and update your `biallelic.py` function for readability purposes (also once you open a file you don't need to open it again)
- [x] run all of this on chromosome 20 and if everything looks good then run on chromosme 1 (currently I believe bcftools and your parser will not produce identical VCFs) 
- [x] create a `run_me.sh` which shows how to run all of the code on oscar

briankirz commented:
Question about SLURM jobs:
What I got from our meeting yesterday was that I was doing computational work locally, while I should have been shell files/SLURM scripts in order to submit the job to OSCAR.
My understanding is that this is done by including #SBATCH parameters in the header. However, in your get1KsampleFile.sh example, I'm not noticing any header parameters for input.
If I want to execute the same code on two different files (ie, chromosome 1 and chromosome 20), how would I indicate their different input files? Would I have to change the job manually each time I do it? I searched "input" in the full SLURM documentation but all I found was something related to `sattach` and I feel like I'm going down the wrong road again
<img width="728" alt="Screen Shot 2022-04-14 at 2 56 10 PM" src="https://user-images.githubusercontent.com/38798045/163457986-cbfa80c0-4423-466f-86db-568f2cba4400.png">
.

briankirz commented:
I'm wondering about how to run input into the SLURM job.
I want to load bcftools and then execute the command on a specific file. Here's what I've got so far
<img width="841" alt="Screen Shot 2022-04-14 at 3 15 42 PM" src="https://user-images.githubusercontent.com/38798045/163460647-ab129c1e-d302-473b-bce5-ea3296a443f4.png">
I found --input=<name> in the SLURM documentation but I'm not sure how to use the name of that file. Will I need to hardcode a different SLURM job for the same command being run on different files?

Maybe I could take a look at the more complex SLURM job list that you showed me during our last meeting in order to look for clues?

briankirz commented:
"also once you open a file you don't need to open it again" as it relates to biallelic.py
Not sure what you mean by this

David-Peede commented:
> 

Look back through my tutorial on OSCAR. The SLURM header is just to request resources NOT EXECUTE CODE. You are still writing in python but bcftools is a command line tool much like doing `cd` or `pwd`

David-Peede commented:
> "also once you open a file you don't need to open it again" as it relates to biallelic.py Not sure what you mean by this

`with gzip.open(vcf_input, 'rb') as gz, open(vcf_output, 'w') as f:` you only need the first open

briankirz commented:
When I compare vcfs in compare_vcf.py, should I be preparing for the headers of each file to not necessarily be the same length? This is taking longer to implement, as I see that the file created by a SLURM job has a different header than the one created by my python code.

It would be much easier to implement if both files were identical (including headers), but I'm not sure that there's a way to do this where I can assume that.

David-Peede commented:
> When I compare vcfs in compare_vcf.py, should I be preparing for the headers of each file to not necessarily be the same length? This is taking longer to implement, as I see that the file created by a SLURM job has a different header than the one created by my python code.
> 
> It would be much easier to implement if both files were identical (including headers), but I'm not sure that there's a way to do this where I can assume that.

Just skip any line starting with a `##` or `#` and there shouldn't be any problems

briankirz commented:
Two things:
1) including the -I (exclude indels) argument in the second command results in an output that is blank besides the SLURM header
<img width="769" alt="Screen Shot 2022-04-15 at 6 28 10 PM" src="https://user-images.githubusercontent.com/38798045/163649405-3dd59efc-3923-4db4-81f3-d2b3043440a1.png">

However, without the -I it works fine. I'll keep experimenting

briankirz committed:
push to rebase

briankirz committed:
VCFs: Comparing bcftools filtering

This is a push of my python functions and shells for filtering the indels out of vcf chromosomes with different bcftools methods, then comparing the results in various ways. Also includes a README in markdown that explains what each function does. Also includes folders with the .txt file results of the filtering and comparisons, as well as some sanity-check testing files.

David-Peede committed:
Merge branch 'main' into Processing-VCF-files-checklist

David-Peede committed:
Update README.MD

David-Peede commented on kirz/vcf_processing/README.MD:
```diff
@@ -0,0 +1,46 @@
+# Filtering chromosomes to include only biallelic SNPS
```
- [x]  brief description of the purpose of this repo
- [x]  brief description of the code scripts including input and output description 
- [x] brief description of the other directories and what they contain
- [x] step by step example of how to recreate the analyses

David-Peede replied:
- [ ] fix bash one liners

David-Peede reviewed (changes requested):
clean up file names and code

briankirz committed:
Cleaning up README and directory

README now includes:
- brief description of the purpose of this repo
- brief description of the code scripts including input and output description
- brief description of the other directories and what they contain
- step by step example of how to recreate the analyses

briankirz committed:
Merge branch 'Processing-VCF-files-checklist' of https://github.com/David-Peede/mentee_research into Processing-VCF-files-checklist

briankirz committed:
Pushing to Update

Processing VCF file branch erroneously contains the summer hmm stuff. Fixing this soon.

David-Peede commented on kirz/vcf_processing/bcf_biallelic.sh:
```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
```
- [x] remove this file

David-Peede reviewed (changes requested):
- [x] fix readme
- [ ] remove unnecessary files

briankirz committed:
July 18 Update Push

Making things recent so I can restructure the files

briankirz committed:
Readme made 1liners

briankirz committed:
Resumbit

Caps Change

briankirz committed:
Moving HMM off-branch

briankirz committed:
Deleting Test Doc
